File: manager/src/test/java/com/revature/end_to_end_tests/pages/LoginPage.java
package com.revature.end_to_end_tests.pages;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class LoginPage extends BasePage{

    @FindBy(id="username")
    private WebElement usernameField;
    @FindBy(id="password")
    private WebElement passwordField;
    @FindBy(css="button[type='submit']")
    private WebElement loginButton;

    public LoginPage(WebDriver driver){
        super(driver);
        this.wait = new WebDriverWait(driver, Duration.ofSeconds(DEFAULT_TIMEOUT));
        PageFactory.initElements(driver, this);
    }

    public void enterUsername(String username) {
        wait.until(ExpectedConditions.visibilityOf(usernameField));
        usernameField.clear();
        usernameField.sendKeys(username);
    }

    public void enterPassword(String password) {
        wait.until(ExpectedConditions.visibilityOf(passwordField));
        passwordField.clear();
        passwordField.sendKeys(password);
    }

    public DashboardPage clickLogin() {
        wait.until(ExpectedConditions.elementToBeClickable(loginButton));
        loginButton.click();
        return new DashboardPage(this.driver);
    }

    public LoginPage clickLoginExpectingError(){
        wait.until(ExpectedConditions.elementToBeClickable(loginButton));
        loginButton.click();
        return this;
    }

    public DashboardPage login(String username, String password){
        enterUsername(username);
        enterPassword(password);
        return clickLogin();
    }
}

File: employee/tests/end_to_end_test/pages/login_page.py
from selenium.webdriver.common.by import By

from tests.end_to_end_test.pages.base_page import BasePage


class LoginPage(BasePage):
    USERNAME_FIELD = (By.ID, "username")
    PASSWORD_FIELD = (By.ID, "password")
    LOGIN_BUTTON = (By.CSS_SELECTOR, "button[type='submit']")
    LOGIN_MESSAGE = (By.CSS_SELECTOR, "#login-message p")

    def open(self):
        super().open()

    def enter_username(self, username):
        self.type(self.USERNAME_FIELD, username)

    def enter_password(self, password):
        self.type(self.PASSWORD_FIELD, password)

    def click_login_button(self):
        button = self.wait_for_clickable(self.LOGIN_BUTTON)

        button.click()

        from tests.end_to_end_test.pages.dashboard_page import DashboardPage
        return DashboardPage(self.driver)

    def clear_username(self, username):
        pass

    def login(self, username, password):
        self.enter_username(username)
        self.enter_password(password)
        self.click_login_button()
        from tests.end_to_end_test.pages.dashboard_page import DashboardPage

        return DashboardPage(self.driver)

